File: Lab Assignment 18/Task 5/java/TestCar.java
public class TestCar {
    public static void main(String[] args) {
        // Test 1: Create Toyota car and check attributes
        Car car1 = new Car("Toyota", "Corolla", 2020);
        System.out.println("Test 1 - Toyota car attributes:");
        if (car1.brand.equals("Toyota") && 
            car1.model.equals("Corolla") && 
            car1.year == 2020) {
            System.out.println("✅ PASS");
        } else {
            System.out.println("❌ FAIL");
        }

        // Test 2: Create Honda car
        Car car2 = new Car("Honda", "Civic", 2018);
        System.out.println("\nTest 2 - Honda car attributes:");
        if (car2.brand.equals("Honda") && 
            car2.model.equals("Civic") && 
            car2.year == 2018) {
            System.out.println("✅ PASS");
        } else {
            System.out.println("❌ FAIL");
        }

        // Test 3: Check year is integer type (conceptual test)
        System.out.println("\nTest 3 - Year data type:");
        System.out.println("✅ PASS (year is int type by design)");
        
        System.out.println("\n🎉 All Java Car tests passed!");
    }
}

File: Lab Assignment 18/Task 5/Python/test_car.py
from car import Car

# Test 1: Attribute values are set correctly
car1 = Car("Toyota", "Corolla", 2020)
assert car1.brand == "Toyota"
assert car1.model == "Corolla"
assert car1.year == 2020

# Test 2: Different car object
car2 = Car("Honda", "Civic", 2018)
assert car2.brand == "Honda"
assert car2.model == "Civic"
assert car2.year == 2018

# Test 3: Year is an integer
car3 = Car("Ford", "Focus", 2015)
assert isinstance(car3.year, int)

print("All Python Car tests passed!")
